aptos-move/aptos-vm/src/block_executor/mod.rs

```diff
@@ -242,7 +250,7 @@ impl BlockAptosVM {
                 // Flush the speculative logs of the committed transactions.
                 let pos = output_vec.partition_point(|o| !o.status().is_retry());
 
-                if state_view.id() != StateViewId::Miscellaneous {
+                if !is_sharded_execution && state_view.id() != StateViewId::Miscellaneous {
```

Comment:
@gelash @sitalkedia here it seems we have to not touch the speculative logging global data structure at all, including not initializing and not flushing.